sql/core/src/main/scala/org/apache/spark/sql/execution/command/createDataSourceTables.scala

```diff
@@ -204,13 +194,21 @@ case class CreateDataSourceTableAsSelectCommand(
       case None => data
     }
 
+    val tableWithPath = if (table.tableType == CatalogTableType.MANAGED) {
+      table.withNewStorage(
+        locationUri = Some(sessionState.catalog.defaultTablePath(table.identifier)))
+    } else {
+      table
+    }
+
     // Create the relation based on the data of df.
     val dataSource = DataSource(
       sparkSession,
       className = provider,
-      partitionColumns = table.partitionColumnNames,
-      bucketSpec = table.bucketSpec,
-      options = optionsWithPath)
+      partitionColumns = tableWithPath.partitionColumnNames,
+      bucketSpec = tableWithPath.bucketSpec,
+      paths = tableWithPath.storage.locationUri.toSeq,
```

Comment:
```
tableWithPath.storage.locationUri
```
This patterns also appears multiple times.

Maybe create a helper method in class CatalogTable called location?
```
def location: Option[String] = this.storage.locationUri
```